context.go

```diff
@@ -325,16 +325,15 @@ func (c *context) FormValue(name string) string {
 }
 
 func (c *context) FormParams() (url.Values, error) {
-	if strings.HasPrefix(c.request.Header.Get(HeaderContentType), MIMEMultipartForm) {
-		if err := c.request.ParseMultipartForm(defaultMemory); err != nil {
-			return nil, err
-		}
-	} else {
-		if err := c.request.ParseForm(); err != nil {
+	req := c.Request()
+	if strings.HasPrefix(req.Header.Get(HeaderContentType), MIMEMultipartForm) {
+		if err := req.ParseMultipartForm(defaultMemory); err != nil {
 			return nil, err
 		}
+	} else if err := req.ParseForm(); err != nil {
```

Comment:
Else can be removed, since the above if is returning.

Reply:
nope.. there is one more nested condition

Reply:
I don't get it? Else if makes no sense here and rarely anyway.

331 returns from 330.

If it returns, `else` is not needed. If it doesn't, only `if` is enough.

Reply:
hm..

> If it returns, else is not needed. If it doesn't, only if is enough.

kk, we can change this

Reply:
I don't really understand how the `else if` can be dropped in this context... The way it is now, a request either has  `MIMEMultipartForm` content type and is processed accordingly OR (and this is where `else` comes into play) a simple form will be parsed. If you drop the else, then in case of successful multipart (thus no return on the `if` branch), you will ALSO parse it as a simple form (since, you have no else, and you don't return early either).  So I'm really curious to see how the else could be dropped in this context...

Reply:
my fail 😔 @alexaandru you'll right

Reply:
You're correct.

Reply:
All's well when it ends well :) Now that I think of it a bit more, we can still ditch the else, with a tiny twist ;)

Instead of:

```Go
  req := c.Request()
  if strings.HasPrefix(req.Header.Get(HeaderContentType), MIMEMultipartForm) {
    if err := req.ParseMultipartForm(defaultMemory); err != nil {
      return nil, err
    }
  } else if err := req.ParseForm(); err != nil {
    return nil, err
  }
  return req.Form, nil
```

we could do:

```Go
	req := c.Request()
	if strings.HasPrefix(req.Header.Get(HeaderContentType), MIMEMultipartForm) {
		err := req.ParseMultipartForm(defaultMemory)
		return req.Form, err
	}
  err := req.ParseForm()
  return req.Form. err
```

We can still save two lines, though I'd personally put a blank line after the if, to nicely separate them :) but that's just me.

I was almost gonna suggest something stupid :), this: `return req.Form, req.ParseForm()` and on a local test it even worked! Which would've been shorter by yet another two lines :) But then I dug further and found this: https://github.com/golang/go/issues/3852 - it turns out the order of evaluation is not specified, it could go either way, and we'd rely on an implementation detail, which would be terrible.

Cheers!